src/app/cake/cake.model.ts

```diff
@@ -0,0 +1,39 @@
+import { Schema } from 'mongoose';
+import { CakeStatus } from './cake-status.enum';
+
+export const CakeSchema = new Schema({
+  name: {
+    type: String,
+    required: [true, 'Why no name?'],
+    min: [5, 'Required, 5 characters min'],
+    max: 50,
+  },
+  description: {
+    type: String,
+    min: [50, 'Required, 50 characters min'],
+    max: 100,
+  },
+  ingredients: {
+    type: [String],
+  },
+  price: {
+    type: Number,
+    default: 0,
+    min: 0,
+  },
+  stock: {
+    type: Number,
+    default: 1,
+    min: 1,
+  },
+  status: {
+    type: String,
+    enum: [CakeStatus.Available, CakeStatus.LastUnits, CakeStatus.OutOfStock],
+  },
+});
+
+CakeSchema.path('ingredients').validate(function (val: []) {
```

Comment:
It would be better to move this logic (business rules) to the class, on this way you will have design centered on the domain (Domain-Driven Design).

Reply:
Ok! Perfect :), I have a little question  is better add this in the validate option for the element? For example: 

`customerPhoneNumber: {
    type: String,
    validate: {
      validator: function (v: string) {
        return /\d{3}-\d{3}-\d{4}/.test(v);
      },
      message: (props) => `${props.value} is not a valid phone number!`,
    },
    min: 10,
  }`

Reply:
This validation can be general for everyone, making a component and apply the function can be better :D

Reply:
Perfect!!! Thanks :)